### ISSUE TITLE
Resolve conflict with "Open Source" misuse and identify better terminology

### ISSUE THREAD

**Issue by chadwhitacre** [getsentry/fsl.software#2]
In getsentry/fsl.software#4 we're writing a new license as a first step. The bigger idea is a new brand other than Open Source to represent our values of [user freedom _and_ developer sustainability](https://blog.sentry.io/sentrys-open-source-values/). What might it be?

Update (@dcramer): The intent here is to normalize a category of software which provides more freedom than closed source, and more freedom than typical source-available software. It should not diminish the value of FOSS (or OSI's definition of "Open Source"), but should also not be overly confusing to end-users, and not include prejudice.

**Comment by chadwhitacre** [comment]
Turns out the US DoD has [an opinion](https://dodcio.defense.gov/Open-Source-Software-FAQ/#q-is-there-a-name-for-software-whose-source-code-is-publicly-available-but-does-not-meet-the-definition-of-open-source-software). 😏

> **Q: Is there a name for software whose source code is publicly available, but does not meet the definition of open source software?**
> At this time there is no widely-accepted term for software whose source code is available for review but does not meet the definition of open source software (due to restrictions on use, modification, or redistribution). Terms that people have used include “source available software”, “open-box software”, “visible-source software”, and “disclosed-source software”. (Such terms might include open source software, but could also include other software).

**Comment by chadwhitacre** [comment]
> Keygen is an open, source-available software licensing and distribution API, built for developers, by developers.

Seeing this on https://keygen.sh/ from @ezekg ... "open, source-available" really toeing the line! 😅

**Comment by chadwhitacre** [comment]
> Codecov++ is now _________ ____________

How I think of this is, what are we gonna use for our next [headline](https://about.codecov.io/blog/codecov-is-now-open-source/)?

**Comment by chadwhitacre** [comment]
> functional source - everything a developer needs to use and learn from the software

From private Slack

**Comment by chadwhitacre** [comment]
> i feel any name needs "source" as the 2nd word (needs to feel like an alternative to "open source"), ideally has just 2 words (more memorable, rolls off the tongue, etc), and somehow the first word needs to do a lot of heavy lifting in communicating how this is different ... functional source hits those pretty well

**Comment by chadwhitacre** [comment]
But "Free Software" ... second term "software"?

**Comment by ljharb** [comment]
Any qualifier to "source" or "software" needs to also not be implying anything inaccurate or negative about the other Sources and Softwares out there, I think.

**Comment by chadwhitacre** [comment]
Hrm, I guess "Superior Source" is out then. 😏

**Comment by chadwhitacre** [comment]
https://microsoft.fandom.com/wiki/Shared_source
https://en.wikipedia.org/wiki/Shared_Source_Initiative

Would MS want to participate in a "Shared Source" revival? 🤔

**Comment by chadwhitacre** [comment]
Also quite parallel: https://fair.io/

**Comment by chadwhitacre** [comment]
I am reminded of [Fair Code](https://faircode.io/) (different than Fair Source) by @ezekg in https://keygen.sh/blog/all-your-licensing-are-belong-to-you/. Keygen and Sentry both make the cut! 💃 Seems actively maintained, as well, since Codecov and Hashi are both listed.

<img width="478" alt="Screenshot 2023-10-24 at 9 47 14 PM" src="https://github.com/getsentry/loose-confederation/assets/134455/eb7bc039-ea0e-4fd7-b27e-9072bc3098b2">

**Comment by chadwhitacre** [comment]
Common Source 🤔

**Comment by chadwhitacre** [comment]
First impressions of "Common Source Software"?

For me it is suggestive of "Creative Commons" in a good way—it doesn't overly overlap in the way "Public Source" would overlap with "Public Domain." At the same time it has some real semantic content, since the concept of a commons is quite defensible as what we are building here. It's quite a strong story, actually, in that [_Governing the Commons_](https://www.amazon.com/dp/0521405998) is all about putting some quasi-governmental enforcements in place to prevent overconsumption. There's a strong research grounding for our approach.

"Codecov is now Common Source" Could hunt!

Thoughts from others?

**Comment by chadwhitacre** [comment]
@benvinegar in Slack:

> first instinct: common source is cool
> surprised we didnt think of that

💃

**Comment by chadwhitacre** [comment]
From @gavinzee:

> &lt;gamer reaction&gt; Isn't it better to be legendary, rare or uncommon?
> &lt;practical reaction&gt; common source works thematically with everything a developer needs and with us wanting to build a coalition of like-minded organizations

👍

**Comment by chadwhitacre** [comment]
From private email:

> I really like Sustainable Source or even better Sustain Source software. I feel it resonates better with your values and is more meaningful than Common Source. A Sustain Source software is one that's resilient to the tragedy of the commons.
> 
> That fits well with your own history as one of the founders of the Sustain (movement).

**Comment by chadwhitacre** [comment]
+1 Fair Code from @ssddanbrown in https://github.com/getsentry/loose-confederation/issues/10#issue-1977911531:

> I always thought "Fair Code" sounded good and could do well to become established if pushed by a few.

**Comment by chadwhitacre** [comment]
Oh! Also: [Commons Clause](https://commonsclause.com/) (h/t @ssddanbrown).

**Comment by chadwhitacre** [comment]
Cooperative Source?

Limits [free-riding](https://en.wikipedia.org/wiki/Free-rider_problem), encourages cooperation.

> In the social sciences, the free-rider problem is a type of market failure that occurs when those who benefit from resources, public goods and common pool resources do not pay for them or under-pay. Examples of such goods are public roads or public libraries or services or other goods of a communal nature. Free riders are a problem for common pool resources because they may overuse it by not paying for the good (either directly through fees or tolls or indirectly through taxes). Consequently, the common pool resource may be under-produced, overused, or degraded. Additionally, it has been shown that despite evidence that people tend to be cooperative by nature (a prosocial behaviour), the presence of free-riders causes cooperation to deteriorate, perpetuating the free-rider problem.

https://en.wikipedia.org/wiki/Free-rider_problem

Common Source is still my favorite.

**Comment by chadwhitacre** [comment]
Trying on "Open Software Product Consortium."

> Codecov is now an **Open Software Product**.

> Acme, Inc. joins the **Open Software Product Consortium**.

OSS — Open Source Software
⇒ OSP — Open Software Product

OSD — Open Source Definition
 ⇒ OSPD — Open Software Product Definition

**Comment by ssddanbrown** [comment]
I'd advise avoiding using "open" if possible since it that's more likely to create friction with the open source community.
It would likely be considered open washing and, especially with using the same "OS" initials, it could be seen as an attempt to ride on, or conflate with, "open source".

**Comment by chadwhitacre** [comment]
Fair points. It's not clear to me yet what the right level of provocative-ness will end up being, this option would be on the more provocative end of the spectrum, for sure.

**Comment by astrojuanlu** [comment]
Yeah and "open" doesn't convey

> The bigger idea is a new brand other than Open Source to represent our values of [user freedom and developer sustainability](https://blog.sentry.io/sentrys-open-source-values/).

Does the name have to be an English word that describes exactly what the license conveys though? "Apache 2.0" and "MIT" are specific licenses (unlike BUSL, which are actually a license family or license type), maybe you can allow more creativity in the name.

**Comment by chadwhitacre** [comment]
> Common Source 🤔

TIL

- http://oss-watch.ac.uk/resources/communityvsopen
- https://web.archive.org/web/20150319223914/http://www.nacubo.org/Business_Officer_Magazine/Magazine_Archives/July-August_2007/Mitigating_the_Risks_of_Big_Systems.html

**Comment by chadwhitacre** [comment]
"Open for Business" 🤔

**Comment by astrojuanlu** [comment]
Playing with acronyms: https://allenai.org/impact-license

> The AI2 ImpACT Licenses get their name from AI2’s four core organizational values: Impact, Accountability, Collaboration, and Transparency.

**Comment by chadwhitacre** [comment]
**Source Available** might be the easiest. It's what everyone is already calling these licenses, and there's a pretty good storyline around convergence over time ... Open Source eventually settled into a handful of popular licenses, Source Available is kinda undefined and amorophous now but could be seen as converging on a handful of approaches (open core, DOSP).

**Comment by ljharb** [comment]
Source Available is very broad, and imo covers any license that isn’t OSI-approved but where the source code is public. Adding a qualifier may help, but the term itself wouldn’t likely ever mean “just this one license”.

**Comment by ezekg** [comment]
@chadwhitacre I hate the term Source Available. Most assume with a source available license that you can't fork, can't modify, etc. Generally, Source available has generally meant that the source code is available to read and that's it. The licenses discussed here all allow more than reading, so I feel the term doesn't fit and just makes things more even confusing. All my customers call Keygen open source, and I didn't tell them to call it that, it's just what they decided on from looking at the ELv2 license.

**Comment by chadwhitacre** [comment]
Strong feelings. 😅

**Comment by ssddanbrown** [comment]
>  All my customers call Keygen open source, and I didn't tell them to call it that,

Not saying that's not true, but specifically labelling yourself as "**open, source-available**" may have an influence to customers thinking that. Plus you did label yourself as open source [in your announcement](https://keygen.sh/blog/all-your-licensing-are-belong-to-you/).

---

I agree that source available is already established as quite a broad field. Any reason not to add weight to the existing [Fair Code](https://faircode.io/) efforts? since it already has some other players involved and thus a little momentum. I always thought that term has quite a strong stance, and it's not used elsewhere and/or conflating with existing common terms.

**Comment by astrojuanlu** [comment]
> Fair-code is not a software license. It describes a software model 

Yeah I think "Source Available" also describes a model, rather than a specific license or family of licenses

**Comment by chadwhitacre** [comment]
> community code

([src](https://twitter.com/steveruizok/status/1755011545279426662))

**Comment by ShaneCurcuru** [comment]
"Source Available" would be great as a descriptor, so I'd vote for that.  Of course, I also get some feel the past negative connotations.

"Community Code" I don't personally like, because "Community" should IMO be something the broader group equally shares.  But - On the other hand, "Community Code" is a good description, and uses neither "open" nor "source" as words, so I think it's actually a really good choice for your model here.

Reading the scrollback:

> Hrm, I guess "Superior Source" is out then. 😏

Actually, using that name would be fine with me!  We should all recognize that anything promoting for-profit (even by merely preventing competition with the license) is going to engage in puffery in their branding.

But I agree, it wouldn't be polite.

**Comment by andrew** [comment]
Some interesting names from a chatgpt conversation:

- Shared Source
- SourceFlex (or something around flexibility)
- Adaptable Source
- Community Core
- Grassroots Source
- Indie Source
- Prosper Code

**Comment by chadwhitacre** [comment]
Shared Source [surfaced above](https://github.com/getsentry/fsl.software/issues/2#issuecomment-1710831686). It was Microsoft's response to Open Source waaaay back in 2001, so it carries lots of baggage. Personally I think it would be hugely epic if we ended up circling back to it. Let's get Windows and Office released under the FSL as part of the Shared Source revival! 😆

**Comment by chadwhitacre** [comment]
From @dcramer on #31:

> Aside, I would love if we can just bucket these licenses into COSS. I dont think thats going to sit well with some, but thats really what they are at the end of the day.
>
> Additionally many COSS projects, while thought of as Open Source, are simply not. They're often open core where the meat of the project is closed source or otherwise a proprietary/fully restricted license.
>
> I do think in general tho these kinds of licenses, including the Open Core model, fit very well with the concept of COSS.

COSS stands for Commercial Open Source Software.

**Comment by ssddanbrown** [comment]
I'd strongly advise against COSS/"Commercial Open Source Software". The "Commercial" part is a specifier, to describe a subsection of open source software. I feel like using it in that way would be equivalent to me describing a "lorry" as a "commercial bus". Plus, open source can be commercialised, and is a fundamental right provided by both free and open source software, it's just harder to do in the manner desired in these cases, or for specific business models. I personally don't see why it's valid to widen the term via that prefix, and it feels somewhat discourteous to all the commercial projects that have managed to do well while taking the risks that open source entails.

I realise my input is quite the contrasting view to many in this thread, so I just want to make it clear why I'm passionate about this. I believe there is a clear need for this open-source adjacent space to be defined, and while it's not defined there is a repeated contention and friction. I totally believe in the open source definition, and I don't want to see that become eroded, especially with how much OSD-adhering work has been put out there demonstrating the benefits & potential of that definition, which has built up to form the popularity and reputation open source has today. Defining that adjacent space would not only help create a clear path for other folks in the same shoes as projects like Sentry, but also help firm those boundaries to protect open source to the OSD.

Sentry has resources, an audience, and a reputation, especially in that VC-space where a lot of these issues stem from. There's opportunity to do a lot of good by defining that adjacent space. Otherwise it would pain me to see that effort and resources put into increasing that tension & friction, especially as I've never been able to understand why there's so much sheer desire to use "open source" while simultaneously feeling that the existing OSD does not suit your need.

**Comment by dcramer** [comment]
> Sentry has resources, an audience, and a reputation, especially in that VC-space where a lot of these issues stem from. There's opportunity to do a lot of good by defining that adjacent space. Otherwise it would pain me to see that effort and resources put into increasing that tension & friction, especially as I've never been able to understand why there's so much sheer desire to use "open source" while simultaneously feeling that the existing OSD does not suit your need.

Exactly why we're investing into it, because we want to see it better.

Re: OSD and "Open Source" - this is just what we see in the community. The defense you hear from folks like us is very true: most people only care about free as in beer, not about all the rights associated with it. Its ok if you believe thats not enough (its not at odds), but thats where the confusion comes in. I fundamentally agree we need a better definition, just one that doesnt have a negative/preconceived stigma around it. This however is not just a problem for venture backed companies, its a problem for the whole community. Unfortunately, partially due to the fault of people who want to be included in the Open Source side of things, most of the time the open source community has come off as fairly hostile. That isnt great as a lot of the people the hostility is directed against are funding, building, and shipping a lot of today's generation of software, both open source and not. That has to stop and we have to find a constructive path forward that everyone is aligned with.

My belief is that we need something that when the mainstream developer see it, its either obvious what it means (and it has to imply a lot more than what some terms imply today about non-permissive open source), OR its compelling enough that they will quickly learn what it means. That's why I'm overly focused on this as a branding / marketing exercise, because thats what it is. Its the same as the OSD's marketing investment behind Open Source, and to me this is critical to get right or the industry is going to constantly be fighting about this and no one will win (well, at least the open source community is unlikely to come out better).

**Comment by ezekg** [comment]
I personally find it confusing that COSS, FOSS, and OSS all mean the same thing, according to the OSI and its adherents. I feel there's room in both COSS _and_ OSS to not be strictly FOSS. But maybe that's just me. Regardless, there needs to be real discussion about the fact that "normal" (sometimes [called "uneducated"](https://twitter.com/Dieter_be/status/1754901696377921589)) developers/engineers/coders/etc. don't hold to the strict dogma of FOSS and the OSI in order for them to refer to something as "open source."

According to the OSI and its adherents, we must shoehorn the term "source available" to mean everything not OSI-approved, but the term "open source" can't naturally evolve. But I'm fine with a new term. One without any baggage, and one that will give us all a rest from the recurrent appearances of the fervent advocates wielding their pitchforks in the name of the holy OSI everytime somebody launches a new "open-source-available" product.

**Comment by dcramer** [comment]
> I'd strongly advise against COSS/"Commercial Open Source Software". The "Commercial" part is a specifier, to describe a subsection of open source software. I feel like using it in that way would be equivalent to me describing a "lorry" as a "commercial bus". Plus, open source can be commercialised, and is a fundamental right provided by both free and open source software, it's just harder to do in the manner desired in these cases, or for specific business models. I personally don't see why it's valid to widen the term via that prefix, and it feels somewhat discourteous to all the commercial projects that have managed to do well while taking the risks that open source entails.

Aside, one note here, I dont think COSS is a commonly used term. I hadn't not seen its use before maybe even the last year.

**Comment by ljharb** [comment]
I would agree COSS  is very much not commonly used, but certainly the word "commercial" is.

**Comment by adamhjk** [comment]
I don't think you can call it "commercial open source software", given that it still implies that the software in question is "open source". I get where you're coming from, since the COSS folks have basically watered down their own definition of what an "open source company" is to be "companies". :) 

"Open Future Software"? 

"Protected Source"?

"Innovator Protected Software"?

**Comment by astrojuanlu** [comment]
No mention of Delayed Open Source in this thread #3 

(Not very exciting as a name, I know)

**Comment by astrojuanlu** [comment]
"Cooperative" and derivatives has been proposed a few times, but I don't think it's a good word. At the moment, these licenses are categorized as "non-compete" (see #31) because the original motivation is to, well, restrict competition to sustain a business model. Cooperatives, as legal entities and as a movement, are completely orthogonal to this.

At the same time, I reckon "non-compete licenses" is anything but sexy as a name.

Also, names are just names. "Free Software" has been battling forever with the Free vs Gratis thing (hola, hablo español). "Open Source" as a name is very well known but the thing that anchors it is not the name, it's the Open Source Definition.

My heart so far is with Sustainable Source https://github.com/getsentry/fsl.software/issues/2#issuecomment-1772945964 or Common Source

**Comment by ssddanbrown** [comment]
> [...] If they don't, I think we do the entire community a disservice by trying to separate the two when to a customer they're not all that different.

>  I think anything that doesn't have "open source" as part of it is going to be confusing to the large group of people (we can just start calling them "the uneducated" 😂 ) who understandably interpret that commonly used phrase to be the logical opposite of "closed source"

I don't really understand why there's such a desire to make the term fit those that are already unaware (or "uneducated" or those that don't care) on the OSD and/or history of open source. That segment will exist for any term since that educate step requires effort, otherwise it's easy to jump to assumptions in any shorter term. Tweaking the existing term won't really help if those users don't already care so much, especially as they'll be more likely to not understand the specific of any additional qualifier and therefore reprogate such content as open source anyway. Why is the focus on that audience rather than those that believe in the OSD, and the history of people building the content that has help put reputation behind that term.

This is a chance to create clearer lines between these segments, rather than blurring this field by adopting or extending existing terminology.

**Comment by chadwhitacre** [comment]
> reprogate

I really wanted to learn a new word today but it seems [this ain't it](https://www.dictionary.com/browse/reprogate). :-)

I think whatever we pick, we have to recognize that it's going to take decades to build up the sort of brand equity we see today in the term Open Source. The first "Codecov is now Reprogate Source" headline is going to fall really flat. The next 10 are going to be equally confusing. Maybe after 100 it's feeling less weird, but it's not until 1,000 headlines that it starts to really feel normal. JJ has been [pushing COSS for _years_](https://www.coss.community/) and look how many people on this thread—people who might be expected to have heard of it by now—are hearing about it for the first time. Settle in, folks. It's gonna be a long haul.

**Comment by chadwhitacre** [comment]
> Edit: Changed "reprogate" to "repropagate", thanks chad!

@ssddanbrown Thanks for coming up with a great placeholder name for us! 😁

> My heart so far is with Sustainable Source https://github.com/getsentry/fsl.software/issues/2#issuecomment-1772945964 or Common Source

I could get behind either of those. We either keep "Open" or "Source" or neither. If we try to keep "Open" I think we will always be fighting a rearguard action against charges of "[openwashing](https://en.wikipedia.org/wiki/Openwashing)." I think "Source" is safer to work with. I like "Common Source" because it has the same rhythm as "Open Source." "Sustainable" is more of a mouthful.

CloudBrew is now Common Source
DataDive is now Common Source
CodeCanvas is now Common Source
PixelPond is now Common Source
EchoMail is now Common Source
QuantumQuery is now Common Source
VirtualVenture is now Common Source
StreamSphere is now Common Source
CyberCrafter is now Common Source
InfinityInsight is now Common Source
NovaNet is now Common Source

**Comment by chadwhitacre** [comment]
Lol it started going with the alphabet and then the Greek alphabet. 😂

<details>
<summary>alphabet</summary>
ApexArchives is now Common Source<br>
BioByte is now Common Source<br>
CryptoCloak is now Common Source<br>
DreamDesigner is now Common Source<br>
EtherEngine is now Common Source<br>
FluxForge is now Common Source<br>
GlyphGrid is now Common Source<br>
HorizonHub is now Common Source<br>
InsightInfinite is now Common Source<br>
JoltJournal is now Common Source<br>
KryptoKey is now Common Source<br>
LinkLattice is now Common Source<br>
MysticMap is now Common Source<br>
NexusNode is now Common Source<br>
OmniOps is now Common Source<br>
PrismPath is now Common Source<br>
QuantumQuest is now Common Source<br>
RiftRoute is now Common Source<br>
SparkSphere is now Common Source<br>
TerraTrack is now Common Source<br>
UmbraUnit is now Common Source<br>
VortexView is now Common Source<br>
WarpWeb is now Common Source<br>
XenonX is now Common Source<br>
YieldYarn is now Common Source<br>
ZenithZone is now Common Source<br>
</details>

<details>
<summary>Greek alphabet</summary>
AlphaAccess is now Common Source<br>
BetaBridge is now Common Source<br>
GammaGate is now Common Source<br>
DeltaDrive is now Common Source<br>
EpsilonEcho is now Common Source<br>
ZetaZone is now Common Source<br>
EtaEngine is now Common Source<br>
ThetaTrack is now Common Source<br>
IotaItem is now Common Source<br>
KappaKey is now Common Source<br>
LambdaLink is now Common Source<br>
MuMesh is now Common Source<br>
NuNet is now Common Source<br>
</detail>

**Comment by mon-jai** [comment]
How about just "delayed FOSS license"?

That is quite clear to me: FOSS but delayed.

**Comment by ezekg** [comment]
> So maybe something like "co-open source" (which could stand for either cooperative or commercial), or "pro-open source" (for professional).

Or, hear me out: "commercial-open source."

**Comment by adamhjk** [comment]
I think the product marketing question here is what are the properties you’re trying to convey with the name, and to what audience. “Common Source” is a totally fine name, but it doesn’t convey much of anything about the properties of the licenses you might bring under its umbrella. What’s “common” about them? 

There is real wisdom in saying you should use “open” or “source” but not both together. 

I think the important properties I’ve heard y’all articulate are:

* No competition 
* Our users can collaborate with us on the source code

are there others? Martina Lauchengco talks about “CAST” as a lens for these things - it should be Clear, Authentic, Simple, and Tested. Let’s put tested aside. 

What about “open user” - essentially saying “this is open for our users, but not for other businesses”.  Open user licensed? It’s clear that the intent is to be open for users. It’s simple “open user license”, it doesn’t lead with a negative. I don’t know how authentic it is (I’ve never heard anyone say “open user license”). 

Another is “User Source” - the source is available to our users, but not for every purpose under the sun. It’s probably as clear as “open user”. It’s relatively simple to explain - it’s like open source, but only for users, not businesses. It feels inauthentic, but that might just be from not using it much. 

“Common Source” isn’t very clear - is it like Creative Commons? Common for whom? It’s not very simple - like open source, only the commons is smaller! It’s probably as authentic as any alternatives. 

Are there other things y’all want conveyed? It might help unlock more options.

**Comment by adamhjk** [comment]
Finding a way to lead with why it’s valuable is the guide. What value does the consumer get from the software being licensed in this way?

**Comment by ezekg** [comment]
@adamhjk my project is exclusively for businesses, as is almost every other project in this conversation, so your "Open User License" term makes no sense to me. My project is for businesses, not users, and it allows everything that more permissive licenses allow except direct competition as a Software-as-a-Service provider. Fork it, modify it, redistribute it, host it, w/e — it's all allowed except for competing as a service provider. Saying these licenses are "open for our users, but not for other businesses" signals you don't actually understand the licenses or projects being discussed, which is a recurring theme in these discussions.

**Comment by adamhjk** [comment]
@zeke I’m trying to help. I’ve been actively involved in trying to have a constructive conversation. You may not like my suggestion, but it’s not coming from a place of ignorance or malice. 

For you, what is the value of being open in this way for those businesses? So they can run it locally and not pay you? I’m not asking like it’s a trap. I’m asking to try and help explore what a good name would be.

**Comment by astrojuanlu** [comment]
Throwing Commercial Source onto the ring 👀

**Comment by adamhjk** [comment]
What about “open non-compete”? It’s clear - it’s open, for everyone, but not to compete with the primary author. It’s simple-ish - non-compete is a bit of a mouth full. It’s authentically what folks are saying they want.

**Comment by adamhjk** [comment]
Commercial Source - simple, in that it fits together well and ties in the commercial aspects of the restriction. It’s not very clear though - we’re explicitly saying you *cant* use it commercially unless you’re the primary/original author.

**Comment by adamhjk** [comment]
I mean, you can use it *in* commerce, but not *for* commerce.

**Comment by ezekg** [comment]
> @zeke I’m trying to help. I’ve been actively involved in trying to have a constructive conversation. You may not like my suggestion, but it’s not coming from a place of ignorance or malice.
> 
> For you, what is the value of being open in this way for those businesses? So they can run it locally and not pay you? I’m not asking like it’s a trap. I’m asking to try and help explore what a good name would be.

I know you've been involved. I've been following this conversation and other conversations (which tend to err more on the side of witch-hunts for the authors of the "offending" projects in the name of the OSI) like it for years, and it's frustrating to have conversations with people who fundamentally don't want to even read or wish to understand the licenses in question. Your comment about the licenses not being for businesses signaled to me that you _don't_ actually understand the license text, or purpose. But I could be wrong, and it may have been a knee-jerk reaction, and if so, I apologize.

It's just that nearly every time I talk to somebody about these licenses (FSL, BUSL, ELv2, SSPL), they think they _only_ grant read-only rights, because that's what "source-available" implies. They don't know you can host, fork, contribute, redistribute, etc. because the term we're told to use by the OSI and its adherents is loaded with ambiguity, so a lot just go with "open source" because it's clear to nearly everyone, even if it steps on some "educated" people's toes.

Regardless, here's why _I_ open sourced Keygen, taken from a HN comment I made a couple weeks ago:

> I wrote about it [a lot more here](https://keygen.sh/blog/all-your-licensing-are-belong-to-you/), when I open sourced my business last year.
> 
> But if you don't want to read the entire post, here's the tl;dr:
> 
> 1. My (larger) customers wanted the source code for auditing purposes. I was using code escrow for larger customers and it was annoying to manage. If some can have the code, why not all?
> 
> 2. My customers wanted me to eliminate my bus-factor of 1. If I got hit by a bus, the business died. Now it can live on and I can grow on my terms.
> 
> 3. My customers wanted to have an "out" if the business ever decided to shut down. Now they have that "out" via self-hosting/forking.
> 
> 4. My customers wanted me to show them they can trust me with their vital business data, and open source builds trust.
> 
> 5. I was tired of dealing with copycats (some verbatim stealing my API schema and docs). Hopefully, moving forward, nobody will want to use a closed source copycat of an open source project.
> 
> 6. My (larger) customers wanted to self-host for compliance purposes (mostly i.r.t. where data is hosted and accessed). If some can self-host, why not all?
> 
> As you can see, most of my reasons (but not all) stem from what my customers wanted. I didn't open source my SaaS for community, or to get free work, or to tap into "a bigger pool of effort." I did it because being closed source was a major pain point for me and my customers.

And to add — I love having community. Even if the community is small. And even if very few people contribute to Keygen, that's fine. The company is the main contributor and driver, anyways. Like I said, I didn't open source Keygen for free work. I did it so that I can communicate freedom, trust, and safety to potential customers and users.

Ref: https://news.ycombinator.com/item?id=39132035

**Comment by chadwhitacre** [comment]
> The company is the main contributor and driver, anyways.

This is pretty key, I think. "Single-source" projects like Sentry and Keygen are different than "classic" Open Source projects like Kubernetes or React. We're talking about our _core products_, not ancillary libraries and frameworks.

**Comment by astrojuanlu** [comment]
Vendor-driven open source projects can evolve into multi-stakeholder, open-governance projects though. And that's orthogonal to the license as far as I understand?

**Comment by ezekg** [comment]
> Vendor-driven open source projects can evolve into multi-stakeholder, open-governance projects though. And that's orthogonal to the license as far as I understand?

Which _core products_ have ever evolved into that, though? Terraform, K8s, React are all ancillary projects, not their core offering.

**Comment by schacon** [comment]
I think what most of us are trying to do with this phrasing is communicate that the repository is public on GitHub and we're actively maintaining it and you can use it for _basically_ anything you would reasonably want to do without being malicious.

A few weeks ago I found a bug in a Planetscale library. It was open source, so I was able to figure out what the problem was and [fix it](https://github.com/planetscale/planetscale_rails/pull/22). So I had the fix. It's nice that they pulled it in too, but the fact that I could inspect and fix it was what I really cared about. Everyone wins. 

That project happens to be under Apache, but _who cares_? For my purposes, it could have been under literally any license and I don't really care. Not being able to _compete_ with them is not even on my radar. It's not on almost _anyone's_ radar for application source code.

For _libraries_ it's entirely different. Because I want to know that if I pull in a crate or a Gem, it's not infected somehow. That could be FSL, it could be GPL. The lines for libraries are drawn between "does a lawyer need to look at this or not". But OSI-approval on a license does not solve that problem for me, so the definition is valueless to me from a pragmatic standpoint. I _still_ have to care what the license is, even if it's OSI-approved "open source"^tm.

I think we're not talking about how much it makes a difference if the thing we're making public is an application or something you can bundle or link to.

**Comment by chadwhitacre** [comment]
Okay, so practically speaking @schacon @ezekg @dcramer yinz three are the ones leading companies who actually need a new brand here, yes? Please chime in if I'm wrong, but I think everyone else on this thread has more of an advisory voice.

Practically speaking, I think we should focus on what the three of you want to create. 

> The intent here is to normalize a category of software which provides more freedom than closed source, and more freedom than typical source-available software. It should not diminish the value of FOSS (or OSI's definition of "Open Source"), but should also not be overly confusing to end-users, and not include prejudice.

What do you envision? A reprogatesource.org website that lays out a few distinctions and licenses?

- permissive
- copyleft
- open core
- delayed open

Something to link to from the next headline? "GitButler is not Reprogate Source." What are the outputs that would achieve the outcomes you three want?

**Comment by adamhjk** [comment]
> > @zeke I’m trying to help. I’ve been actively involved in trying to have a constructive conversation. You may not like my suggestion, but it’s not coming from a place of ignorance or malice.
> > For you, what is the value of being open in this way for those businesses? So they can run it locally and not pay you? I’m not asking like it’s a trap. I’m asking to try and help explore what a good name would be.
> 
> I know you've been involved. I've been following this conversation and other conversations (which tend to err more on the side of witch-hunts for the authors of the "offending" projects in the name of the OSI) like it for years, and it's frustrating to have conversations with people who fundamentally don't want to even read or wish to understand the licenses in question. Your comment about the licenses not being for businesses signaled to me that you _don't_ actually understand the license text, or purpose. But I could be wrong, and it may have been a knee-jerk reaction, and if so, I apologize.

You are wrong, and it was a knee-jerk reaction, and it's completely understandable. I get the frustration. I'm not on a witch hunt. It's all good, friend. :) 

> It's just that nearly every time I talk to somebody about these licenses (FSL, BUSL, ELv2, SSPL), they think they _only_ grant read-only rights, because that's what "source-available" implies. They don't know you can host, fork, contribute, redistribute, etc. _for business purposes_ because the term we're told to use by the OSI and its adherents is loaded with ambiguity, so a lot just go with "open source" because it's clear to nearly everyone, even if it steps on some "educated" people's toes.

I don't think they only grant read-only rights. I think it is very unclear how the derivative work rights they grant, for example, could be used in practice (which is not the same as saying they couldn't be used!) - but that's a separate issue.

In my suggestion above, I used "users" because I was trying to get at the value proposition. Who is it open for, and what can they do with it, and what can't they? What they *can't* do is compete with the original author, pretty much everything else is fair game (even if it might be murky). 

> Regardless, here's why _I_ "open sourced" Keygen, taken from a HN comment I made a couple weeks ago:
> 
> > I wrote about it [a lot more here](https://keygen.sh/blog/all-your-licensing-are-belong-to-you/), when I open sourced my business last year.
> > But if you don't want to read the entire post, here's the tl;dr:
> > 
> > 1. My (larger) customers wanted the source code for auditing purposes. I was using code escrow for larger customers and it was annoying to manage. If some can have the code, why not all?
> > 2. My customers wanted me to eliminate my bus-factor of 1. If I got hit by a bus, the business died. Now it can live on and I can grow on my terms.
> > 3. My customers wanted to have an "out" if the business ever decided to shut down. Now they have that "out" via self-hosting/forking.
> > 4. My customers wanted me to show them they can trust me with their vital business data, and open source builds trust.
> > 5. I was tired of dealing with copycats (some verbatim stealing my API schema and docs). Hopefully, moving forward, nobody will want to use a closed source copycat of an open source project.
> > 6. My (larger) customers wanted to self-host for compliance purposes (mostly i.r.t. where data is hosted and accessed). If some can self-host, why not all?
> > 
> > As you can see, most of my reasons (but not all) stem from what my customers wanted. I didn't open source my SaaS for community, or to get free work, or to tap into "a bigger pool of effort." I did it because being closed source was a major pain point for me and my customers.
> 
> And to add — I love having community. Even if the community is small. And even if very few people contribute to Keygen, that's fine. The company is the main contributor and driver, anyways. Like I said, I didn't open source Keygen for free work. I did it so that I can communicate freedom, trust, and safety to potential customers and users.

Fantastic. The word "customer" stands out for me in what you wrote. Is there something there we can use to think about naming? "Open Customer"?

**Comment by chadwhitacre** [comment]
> "Open Customer"

I don't think we should use "Open," because it'll forever mire us in charges of "openwashing."

**Comment by adamhjk** [comment]
@chadwhitacre fair enough. One problem I've had in brainstorming possible terms is finding one that expresses the clear intent for "open-ness" that is present. It's clear that being "open" really matters to both the primary stakeholders on this thread and their articulation of what matters to the folks they're trying to reach with this approach. There are just very, very few synonyms for open. While the "Open Source" crowd is rightfully protective of what that phrase means, I don't think that should stop y'all from saying specifically that being "open" is a core value you have. I hear it strongly when y'all articulate it.

One advantage of articulating your own set of values and naming them is that you can let go of the need for the existing phrases to embrace you. Some number of folks aren't going to like anything you come up with, even if it was "source available" or "non-compete license", because they have other agendas to push, pet peeves to beat you up with, or whatever. Maybe forget about those people. 

The software is "open", but it isn't "open source". But I think it could be "Open X", as long as 'X' embraced how it is different from Open Source. Y'all do you, but I wouldn't be so quick to remove it from contention, since it's so core to how you (and presumably the people who value what you've done) think about what you're doing.

**Comment by adamhjk** [comment]
What about "Core Source"?

**Comment by dcramer** [comment]
Just for clarity folks - I dont intend to brand FSL under its own unique snowflake terminology. My intent is to focus the conversation around _all forms_ of software which aren't open source, which includes a large amount of open core. This matters because many of those businesses effectively make a similar compromise to FSL (and some other non-compete variants), and is even more important because the approach of the FSL to many users has _more_ value than the approach of hyper restrictive open core (think where e.g. 50% of the product is proprietary, or the key features). Otherwise we'd just call it non-compete and be done with it.

There may not be a clean solution, but I think its important that the community separates FOSS from semi-open source projects, otherwise there's no real defense on why open core gets plastered as open source and something like FSL doesn't. Its also worth knowing that no one will care about this except us - end users will still just care that the software is primarily free, so its important to ensure that remains obvious to people.

Will try to dig more into this as I get time but 🎿ing this weekend.

**Comment by schacon** [comment]
This thread is one of those things that just obsesses me. Like the stay awake at 3a trying to figure out solutions. So dumb.

Anyhow, thinking about it more, I'm liking "Pro Source" the most. A "Pro Source License". 

I do like deviating from "open" and concentrating on "source", but with a positive semantic spin. "delayed" is not positive. "available" is not positive. Perhaps we capitulate on the special meaning of "open" to not mean "opposite of closed", but instead "imbuing 4 inalienable software rights that nobody can name", etc. Fine. "Source" is more important. You get the source code.

We are providing a professionally maintained and stewarded source code for anything reasonable you want to use it for. Pro Source.

"Today X is officially Pro Sourced"

In any case, the most recent of my 3a obsessive branding exercises. 🤣

**Comment by adamhjk** [comment]
@schacon I hear you. It’s a bit of a mind virus. 

I think “Pro Source” is pretty great. It’s clear you’re getting the source code, and that the source code is what matters - that it’s “pro” implies both that it’s good and that it’s professional. It’s authentic - it captures the basic idea that this is source from professional people. It’s simple - by avoiding “open” you won’t get stuck in conversations about the four freedoms.  

I don’t get a “vote” or whatever, but this is a good choice to me.

**Comment by chadwhitacre** [comment]
Thoughts on this as an approach?

https://softwarecommons.com/

**Comment by ssddanbrown** [comment]
@chadwhitacre The core content looks good to me. For a final release I'm not sure how well quoting Wikipedia works (since it could change over time), but maybe you could fix on your own definition with some references to back it up?

I wasn't confident exactly how "commons" is used but creative commons is very popular, and you have licenses like [CC BY-NC](https://creativecommons.org/licenses/by-nc/4.0/) there so I feel the license categories listed fit perfectly well within what you've defined.

As you could probably guess, I'm still not keen on the affirmed desire to use "COSS" as per the notes due to reasons I've given above. I do agree though about how open core _can be_ problematic in terms of how it's marketed, but that's a whole other tangential bucket of worms to get into, and not something I see as a justification to adopt COSS.

**Comment by chadwhitacre** [comment]
To be clear, I would see "Software Commons" as nomenclature we would pursue _instead of_ COSS.

Good call on the Creative Commons parallels.

**Comment by schacon** [comment]
I think to a degree it also comes back to your original question which is "X is now ___ ____". 

Should this be "X is not Software Commons'd?"

I feel like "source" from a semantic point of view is helpful in this area, as it communicates both "source code" and it has the ability to be "sourced".

**Comment by dcramer** [comment]
We have some folks working internally at Sentry on this, so for some transparency I want to send the same feedback I sent there:

Let's focus on the overarching symptoms first, and terminology we can grow out of that. What I mean by that is:

How do we articulate the delta in licenses to end users?

- Permissive
- Copyleft
- Open Core (I'd argue FSL just lives in this chunk, tho we should separate dual-licensed open core vs FSL/BUSL)
- Source-available (this is mostly a term, but folks need education)
- Proprietary/Closed

Thats has to be the goal, and from an FSL agenda: how do we fit into the existing nomenclature? As mentioned I believe we simply put this commercial stuff into the open core category as a whole, but thats unrelated to "Sentry is now FSL" or whatever. Arguably if FSL became mainstream here, and was the right kind of solution, simply saying "Sentry is now available under the FSL" would be enough, but I don't see that happening.

**Comment by chadwhitacre** [comment]
> how do we fit into the existing nomenclature?

One benefit of "software commons" is that [it _is_ existing nomenclature](https://en.wikipedia.org/wiki/Information_commons#Software_commons), and it's pretty darn close to what we want (emphasis added):

> The _software commons_ consists of all computer software which is available at little or no cost and which can be reused with few restrictions. **It includes open source** software which can be modified with few restrictions. However the commons **also includes software outside of these categories** – for instance, software which is in the public domain.

## Parallel with Creative Commons

@ssddanbrown made a really good point about the parallel between Software Commons and [Creative Commons](https://creativecommons.org/). First of all, Creative Commons is a massive brand with a lot of popularity and education. It would be great to draft off of that. Secondly, they allow for restrictions on commercial use and modification while also encompassing the equivalents of Open Source, both permissive (BY) and copyleft (SA):

<img width="480" alt="Screenshot 2024-02-12 at 6 52 49 PM" src="https://github.com/getsentry/fsl.software/assets/134455/2082cb45-086b-4f32-925a-54cb76b51895"><br>

Thirdly, Creative Commons licenses are [famously _not_ intended for software projects](https://creativecommons.org/faq/#can-i-apply-a-creative-commons-license-to-software), which means that the brand space and the Google [results](https://www.google.com/search?q=software+commons) are wide open for us to stake out. Someone is [squatting the .org](http://softwarecommons.org/), but `softwarecommons.com` was available(!).

## Lumping FSL in with Open Core 👎

> put this commercial stuff into the open core category as a whole

[Open core](https://en.wikipedia.org/wiki/Open-core_model) already means something and I don't think we should stretch it to include DOSP, both because brands are difficult to stretch (as we've learned with Open Source ;-), and because it's not particularly positive. [Sid is the only one who loves the term and he comes off as really defensive](https://opencoreventures.com/blog/2023-07-open-core-is-misunderstood-business-model/). Interestingly, Wikipedia considers COSS to be a synonym ("['Open core' and 'COSS' redirect here](https://en.wikipedia.org/wiki/Open-core_model)"). COSS could be seen as JJ's attempt to rebrand Open Core to shed some of the negative associations.

## Benefits of Software Commons for Education

> How do we articulate the delta in licenses to end users?

I think we can do this with Software Commons to encompass sharing practices (some sub-details tbd) and then distinguish Source Available and Closed Source as outside the bounds. Again, we get a lot of education/brand benefit from the Creative Commons parallel, we can probably play this up a lot more and maybe even get an official nod from them.

<img width="774" alt="Screenshot 2024-02-12 at 7 14 06 PM" src="https://github.com/getsentry/fsl.software/assets/134455/1b7cea32-ae27-4f57-8ace-f8f793d7d626">

## The Headlines

> I think to a degree it also comes back to your original question which is "X is now ___ ____".
>
> Should this be "X is not Software Commons'd?"
>
> I feel like "source" from a semantic point of view is helpful in this area, as it communicates both "source code" and it has the ability to be "sourced".

You're not wrong @schacon, but also[ people do use](https://www.google.com/search?q=%22is+now+creative+commons%22) "X is now Creative Commons" and "X is now Creative Commons Licensed." I think we could use "X is now Software Commons" or "X is now Software Commons Licensed" or also "X is now part of the Software Commons."

**Comment by schacon** [comment]
You are starting to convince me. And I like that the domain was available. 😂

**Comment by schacon** [comment]
I wonder, for clarities sake, if we really need 5 groupings though. This is just a thought exercise, so bear with me. I think there are three *major* groups, if grouped by what is important to the developer.

- The permissive license (MIT, Apache, BSD, WTFPL)
    - for any type of project
    - ensures nobody needs to care about anything, really
    - ✅ no lawyers needed
- The copyleft license (GPL family)
    - for any type of project
    - ensures copyleft provisions
    - ⚠️ lawyers need to pay attention
- The business license (BSL, FSL)
    - mainly for applications
    - ensures non-compete protections to dissuade bad actors
    - ⚠️ lawyers need to pay attention

For two of the three, there are legal obligations that need to be paid attention to when including it in your corporate products. For two of three, it has been determined over time that the phrase "open source" is strictly only applicable to those two.

The problem that we've been discussing is that these two pairs do not overlap. Developers are therefore left with the following Venn Diagram of Sadness and Confusion.

![CleanShot 2024-02-13 at 11 13 17@2x](https://github.com/getsentry/fsl.software/assets/70/6d6db69b-7ab2-4b7d-b2c3-f77ddcba6b0c)

Perhaps it would be useful to be able to add variations (like CC-BY-NC) for these three groups, since each one very clearly communicates what the developer needs to care about.

**Comment by chadwhitacre** [comment]
Guys, guys! Check this out! What if we rename the "Functional Source License" to the "Future Source License" and develop "Future Source" as a practice parallel to "Open Core" and "Open Source"?

1. We get to keep the FSL brand we've already started building.
2. We can say "X is now future sourced" if we want to.
3. [Software Commons](https://softwarecommons.com/) is an educational piece relating the different practices.

[Go check out the site](https://softwarecommons.com/), I updated it with the above and I think it's pretty strong. 👍

<img width="819" alt="Screenshot 2024-02-13 at 10 34 54 AM" src="https://github.com/getsentry/fsl.software/assets/134455/8d884239-2fe4-4d23-80bb-9a0bf2684160">

Each of these links goes to an explainer page on the site, so we can educate about each particular sharing practice. Site says:

> Software Commons is focused on education about sharing practices beyond Open Source, particularly [Future Source](https://softwarecommons.com/future-source).

**Comment by chadwhitacre** [comment]
> I wonder, for clarities sake, if we really need 5 groupings though. This is just a thought exercise, so bear with me. I think there are three major groups, if grouped by what is important to the developer.

I think we need to acknowledge Open Core / COSS because it is so well established. [JJ](https://twitter.com/JosephJacks_)/[OSS Capital](https://oss.capital/) with companies like Cal.com etc. are really driving forward that vision, I think we want to have a page we can link to that explains it from our perspective, when it is a good choice and when Future Source is better.

The fifth category is Public Domain—I think the value of keeping that on the chart is that it is referenced in the Wikipedia page and also helps make the point that even permissive licenses do have restrictions (attribution). It's also so common and people know about it, I think we want to be authoritative and comprehensive. The point is to educate folks on the whole field and be a fair though opinionated voice.

**Comment by chadwhitacre** [comment]
@ezekg Keygen [counts for Open Core](https://github.com/keygen-sh/keygen-api?tab=readme-ov-file#software-licensing-for-everyone), yes?

> Keygen comes in two editions. Keygen CE is our Community Edition, and is free (as in beer) to self-host for personal and commercial use. Keygen EE is our Enterprise Edition, and it requires a license key to use.

**Comment by ssddanbrown** [comment]
> Keygen [counts for Open Core](https://github.com/keygen-sh/keygen-api?tab=readme-ov-file#software-licensing-for-everyone), yes?

I wouldn't have considered it open core, since the core is still under a non-open license with limits to modification and usage.

**Comment by selviano** [comment]
I don't like "Future Source" as a category. The best part about FSL is that everything—everything!—is available today. Not just some features, but the full product. The only compromise is that you can't compete with the company licensing under FSL/BuSL, at least not with the latest version. That's a pretty fair compromise, generous even. 

In the case of Sentry, the product is useful for any company that has a developer on staff. Of these companies, I'd venture that 99.99999% of them don't intend to compete with Sentry. 

So the key problem with the term "Future Source", is it emphasizes the _one limitation_ (which, btw, only applies to 0.0000001% of users), and not the immense benefit of being effectively open for 99.99999% of potential users right away.

**Comment by chadwhitacre** [comment]
> it emphasizes the one limitation (which, btw, only applies to 0.0000001% of users), and not the immense benefit of being effectively open for 99.99999% of potential users right away.

Fair point. 🤔

Maybe "Single Origin" is better after all, emphasis on the characteristics of the community around the project vs. the time/space difference.

**Comment by chadwhitacre** [comment]
Open Core companies like to emphasize the external contributions to the core of the product, Single Origin less-so.

**Comment by mswilson** [comment]
A "[Commons](https://en.wikipedia.org/wiki/Commons)" is a specific thing in the literature. It's something where multiple people benefit from resources that are accessible to all.

In the realm of software development (and other cooperative efforts), there is a studied phenomenon called "[commons-based peer production](https://en.wikipedia.org/wiki/Commons-based_peer_production)." The goods that are produced at the center of commons-based peer production are available for the benefit of many stakeholders.

One problem with attaching a "Common Source Software" label onto goods that are made available under FSL is that the commercial benefits accrue to the _sole producer_ of that good. Those goods aren't even [club goods](https://en.wikipedia.org/wiki/Club_good). They don't enrich the public commons of software until 2 years after release.

It's [Freeware](https://en.wikipedia.org/wiki/Freeware) with accompanying source code, until the time when it is available under a FOSS license.

**Comment by dcramer** [comment]
Yall this thread isnt really getting anywhere. Step back and focus on the differences, we'll figure out some magical term later.

- Single Origin is an attribute, it describes nothing.
- Future Source is a random term, still mostly describes nothing. Its Present Source. Its Source.
- Common Source who knows but its not the p0 at the end of the day.

> One problem with attaching a "Common Source Software" label onto goods that are made available under FSL is that the commercial benefits accrue to the sole producer of that good. Those goods aren't even [club goods](https://en.wikipedia.org/wiki/Club_good). They don't enrich the public commons of software until 2 years after release.

Aside, this ignores the fact that the industry benefits greatly from Sentry's "freeware", so Commons is still fitting. It doesnt matter if "Sentry commercially benefits", becausae we've made the software free (and much of it permissive). So suggesting the benefits accrue to just us is wrong, and I'd argue that Sentry being free, rather than say Open Core and restrictive, is actually far more beneficial to the technology industry and humanity in general.

That said, I'm not convinced overlapping w/ Creative Commons (or the Commons Clause) is a good idea, but I'm also still of the mindset that this is not the critical problem needing solved today.  To me, the easier solve is: what is the delta between FSL and Open Core, and is there a broader bucket term that we can label the _entire_ source-available industry under? or at least FSL and Open Core's approaches?

**Comment by mswilson** [comment]
> what is the delta between FSL and Open Core, and is there a broader bucket term that we can label the entire source-available industry under? or at least FSL and Open Core's approaches

100% agree with this focus. The FSL approach that doesn't hold back essential functionality **is better**. Finding a way to highlight this distinction relative to the "historical open core" business model is a good place to spend some effort.

**Comment by astrojuanlu** [comment]
The delta is... big? FSL is a license (well, two). Open Core, as far as I understand it, is a business model in which there are, at least, two products involved: the Core and... the Crust. Nothing dictates what license should the Core have (you could do Open Core _and_ FSL! or AGPLv3, or WTFPL, doesn't matter), and it's assumed that the Crust is proprietary ([Grice's cooperative principles](https://en.wikipedia.org/wiki/Cooperative_principle)).

I don't know, FSL and Open Core are maybe somewhat similar in _intent_ (companies trying to have _some_ open source while keeping the business viable) but I wouldn't consider them in the same category, doing a diff between the two seems complicated IMHO.

**Comment by chadwhitacre** [comment]
> the commercial benefits accrue to the _sole producer_ of that good

Here's where it gets interesting: [individuals have a substantial degree of freedom to participate in the commercial benefits of Sentry](https://cra.mr/open-source-and-a-healthy-dose-of-capitalism):

> Since as early as Sentry started generating revenue we started investing it back into the project. Not on our own salaries, but on giving back to the community in a variety of ways. One of those ways was to identify people who were helping the project and offer them a running contract. That is, if we saw someone contributing in a meaningful way to one of our repositories, we’d offer them a standing hourly contract paying them for their time. We didn’t need to do this, but we recognize they value they added, and in some cases these were serious time commits that people were voluntarily spending.

I've experimented with a much stronger implementation called [take-what-you-want compensation](https://gratipay.news/sharing-our-take-what-you-want-story-911dad62ac32), but even Sentry's current model where people can voluntarily start making PRs to our SDKs and have a real path to sharing in the commercial benefits that accrue is, I would argue, a meaningful and interesting counterpoint. In fact, this is how I applied for my own job at Sentry: [I started making PRs](https://github.com/getsentry/self-hosted/pulls?q=is%3Apr+author%3Achadwhitacre+created%3A%3C2020-11-09).

### I Like It Still

The more I sit with Software Commons, the more it feels like the right place to locate the conversation, because the two failure modes of a commons are [enclosure](https://en.wikipedia.org/wiki/Enclosure) and the [tragedy of the commons](https://en.wikipedia.org/wiki/Tragedy_of_the_commons). Either too many restrictions, or too few. This seems to quite precisely mirror our desire to [balance user freedom and developer sustainability](https://blog.sentry.io/sentrys-open-source-values/). There is much to hammer out in terms of details, but the fundamentals seem sound to me. Exciting! 😁 💃

**Comment by bmann** [comment]
Came here from your blog post Chad. Have been 

I’d say Software Commons ia good and in fact extremely broad in a way that I wouldn’t restrict it merely to “licensing” -  but rather also organizational forms and ways of governing.

**Comment by WisTex** [comment]
I am curious where plugins and addons fall under these categories. Many times the core and the plugins do not have the same licenses. Some examples:

Core: Proprietary (pay to download)
Plugin: MIT License (free to download)

Core: AGPL (free to download)
Plugin: Proprietary (pay to download)

Core: MIT License (free to download)
Plugin: AGPL (free to download)

Core: AGPL (free to download)
Plugin: AGPL (pay to download) - you see this a lot with WordPress themes and plugins.

When I think of open core, the first thing that comes to mind is a freemium offering, where some of the software is free and some of it requires payment. Is that what you mean here?

And another issue with open core is that any software that has a plugin or addon system could theoretically be classified as open core, since, typically, third-parties may license their plugins however they want. Is there really a functional difference between the original developer offering a proprietary plugin and a third-party doing the same?

**Comment by WisTex** [comment]
### Licensing vs. Business Model

Are the categories supposed to be license types or business models?

Instead of trying to combine both into one classification list, perhaps we classify software using both.

Examples:

Licensing: Open Source - Copyleft
Business Model: Freemium (Open Core / Premium Options)

Licensing: Open Source - Permissive
Business Model: Donations & Contributions

Licensing: Public Domain
Business Model: Public Domain

**Comment by chadwhitacre** [comment]
Thanks for chiming in @bmann @WisTex! Thanks for starting #34 and thanks for the links, looking forward to digesting these.

**Comment by chadwhitacre** [comment]
[Above](https://github.com/getsentry/fsl.software/issues/2#issuecomment-1936504225):

> I don't think we should use "Open," because it'll forever mire us in charges of "openwashing."

[Counterpoint](https://redmonk.com/sogrady/2024/02/26/ai-open-source/):

> Open is good. The industry succeeds and is driven forward when groundbreaking new models are released and made available. But for the health of open source and the industry as a whole, it’s important to choose our words carefully and to understand that while open is good, open is not open source.

Still leaning against "open," but [open products](https://gratipay.news/open-products-25824a325fea) though. 🤔

**Comment by chadwhitacre** [comment]
All ours, team. All ours. 😳

https://softwarecommons.com/
https://github.com/softwarecommons
https://twitter.com/softwarecommons
https://mastodon.social/@softwarecommons
https://www.facebook.com/softwarecommons
https://www.linkedin.com/company/softwarecommons

**Comment by RichardLitt** [comment]
I think Software Commons works well, Chad, for the reasons you describe. I like it.

**Comment by chadwhitacre** [comment]
Alright, optics gaffs notwithstanding, it feels to me like we are converging on Software Commons as the new brand we are going to work with to elevate developer sustainability to a value on par with user freedom. Last call before I close this out. 🙌

**Comment by MattiSG** [comment]
Just discovering this initiative today, so not sure this is welcome, but just in case: while I love the initiative and intend to participate, I feel pretty uncomfortable with the name. [Digital Commons](https://en.wikipedia.org/wiki/Digital_commons_%28economics%29) is a clear concept (democratic governance on informational assets that are non-rival and non-exclusive, drawing on [Elinor Ostrom](https://en.wikipedia.org/wiki/Elinor_Ostrom)’s work on Commons), and it has some traction at a political level, especially in [France](https://www.diplomatie.gouv.fr/en/french-foreign-policy/digital-diplomacy/team-blog/digital-commons) and [in the EU](https://ngiforum2023.eu).

Calling this initiative “Software Commons” when it is about sustainability and creating temporary exclusive access will create a major inconsistency and make it very hard to reconcile with the growing number of initiatives that aim at supporting Digital Commons (such as [Next Generation Internet Commons](https://commons.ngi.eu) announced just last month).

**Comment by chadwhitacre** [comment]
You're definitely welcome to chime in, @MattiSG. :-)

Thanks for the links re: digital commons. On the one hand I think "software commons" is distinct enough from "digital commons" that we don't need to line up 100% with how the latter term is being taken up in different contexts. On the other, I see a degree of imprecision in how we are using terms. Common pool resources are different than public goods. The latter are non-excludable and non-rivalrous. The former are precisely rivalrous. Do you know why "informational assets that are non-rival and non-exclusive" are referred to as "digital commons" rather than "digital public goods"?

Ostrom's framework for developing institutions that prevent tragedies of commons ([_Governing the Commons_](https://www.amazon.com/dp/1107569788), p. 90) is precisely about developing and enforcing rules for who may take resources from the common pool, and under what conditions. Successful CPRs collectively evolve norms of quasi-exclusivity to prevent overuse, enforce by institutions. Moreover, these "[i]nstitutions are rarely either private or public - 'the market' or 'the state.' Many successful CPR institutions are rich mixtures of 'private-like' and 'public-like' institutions defying classification in a sterile dichotomy" (p. 14).

I think we're on the right track.

**Comment by chadwhitacre** [comment]
Bringing this over from https://github.com/getsentry/fsl.software/issues/2#issuecomment-1977433883: 

> Ostrom's framework for developing institutions that prevent tragedies of commons ([Governing the Commons](https://www.amazon.com/dp/1107569788), p. 90) is precisely about developing and enforcing rules for who may take resources from the common pool, and under what conditions. Successful CPR management comes from evolving norms of quasi-exclusivity to prevent overuse, enforced by collectively governed institutions. Moreover, these "[i]nstitutions are rarely either private or public - 'the market' or 'the state.' Many successful CPR institutions are rich mixtures of 'private-like' and 'public-like' institutions defying classification in a sterile dichotomy" (p. 14).

**Comment by ohnoitsnoah** [comment]
What is "Sorta-Open Software"

**Comment by ohnoitsnoah** [comment]
Hello!! I'm interested in helping with this project! What is currently in the process of being done for it?

**Comment by joshsimmons** [comment]
Catching up here, whole lotta thoughts and feelings, but the bottom line for me: I appreciate the amount of effort that's gone into trying to come up with a distinct term. While it pains me to see things licensed out of the open source commons, it's worth acknowledging that Sentry has broadly been more honest in its approach to these things.

Wishing you luck in this endeavor.

**Comment by chadwhitacre** [comment]
Thanks for jumping in, @ohnoitsnoah! Now that we've settled on the name, I'm going to close this out and move over to other tickets such as #34 #30 and #27 to revise softwarecommons.com to be more informative. We have #5 for discussion of over-arching goals. I'll probably do some reorganization and shift to the gitthub.com/softwarecommons org.

**Comment by mswilson** [comment]
I'm sorry, but 13 hours is not much time for a "last call."

**Comment by MattiSG** [comment]
> Do you know why "informational assets that are non-rival and non-exclusive" are referred to as "digital commons" rather than "digital public goods"?

The main difference lies in governance models. _Public_ goods in socio-economical theory would be owned and managed by some public institution, whereas _common_ goods are owned by no-one in particular and collectively managed. Now, the evolution of vocabulary is always a tricky thing to explain after the fact, it is no exact science… you know, sometimes some actor falls in love with a term that sounds cool and that they decide to coin quite fast across all platforms in order to ensure they secure it and then they don't want to spend too much time listening to warnings for potential overlaps because that arrives a bit too late for them 😜

For example, “[Digital Public Goods](https://digitalpublicgoods.net)” now refers to open source software that follow best practices in terms of interoperability and standards and that have demonstrated their ability to attain the global SDGs, as certified by the @DPGAlliance (United Nations Development Program and Norad), even though they are not necessarily owned by public actors (as opposed to the “Public Code” that the [Foundation for Public Code](http://publiccode.net/) refers to, for example).

I have shared above some references to the existence of the term “Digital Commons” to designate informational goods with collective ownership and democratic decision-making by its users and contributors. The definition converged internationally for example at the [Open Government Partnership Summit in 2016](http://www.henriverdier.com/2016/12/des-communs-numeriques-pour.html), and you can track the evolution in [scientific literature](https://scholar.google.com/scholar?hl=fr&as_sdt=0%2C5&q=communs+numériques&btnG=) since then, now bubbling up at EU level since the [2022 Digital Assembly](https://digital-strategy.ec.europa.eu/en/events/digital-assembly-2022-closer-look-digital-future) under the French presidency.

The vocabulary is quite loaded today and if you were receptive to these arguments on avoiding “open”, they apply just as well for “commons”, and you're likely to be accused of “commons washing” (it's maybe just less visible from the USA where the term is less prevalent yet):

> I'd advise avoiding using "open" if possible since it that's more likely to create friction with the open source community.
> It would likely be considered open washing and, especially with using the same "OS" initials, it could be seen as an attempt to ride on, or conflate with, "open source".

But most importantly, when re-reading the conversation, I fail to see how “user freedom _and_ developer sustainability” is captured by leveraging the “commons” term: Digital Commons are about having democratic governance over a shared informational asset, by users and contributors. I for one would have a hard time reading that “Public Domain” or “Future Source” are one of five types of “Software Commons”: how exactly does the _community_ assembled around the resource has any say in how to restrict or expand usage rights in these cases? No democratic governance ⇒ no commons.

- - -

Just to clarify: addressing sustainability is critical, I love the ideas of the FSL (and other limited non-compete cases such as [RSAL](https://redis.com/legal/licenses/) or the [CoopCycle license](https://wiki.coopcycle.org/en:license)) and of referencing paths to sustainability. I'm just sharing warnings on naming as this conversation seemed to invite to 🙂

**Comment by dcramer** [comment]
> Calling this initiative “Software Commons” when it is about sustainability and creating temporary exclusive access will create a major inconsistency and make it very hard to reconcile with the growing number of initiatives that aim at supporting Digital Commons (such as [Next Generation Internet Commons](https://commons.ngi.eu) announced just last month).

Just for clarity - Software Commons is not exclusive to this conversation. It’s simply a mechanism to talk about the larger set of values from open software (primarily access), and one which isn’t going to cause confusion around a preexisting registration (Open Source). 

For us it fits the values well of most open source and some other approaches to free software. That means we believe the ideals of FSL are upheld with the commons, but the commons is not intended to mean explicitly the FSL, but a broader set of talking points. 

The aim is to help educate people on the values of the different models, in a way that has unfortunately become impossible around the topic of general open source.

**Comment by chadwhitacre** [comment]
> 13 hours is not much time for a "last call."

Fair enough, reopening. Sorry. 🙇

> The main difference lies in governance models.

I myself am unsettled at using Software Commons without a clearer understanding of [our goals](https://github.com/softwarecommons/softwarecommons.com/issues/2). To me Software Commons only makes sense as a name if we are founding a new institution, which to my mind should be something like a multi-stakeholder platform cooperative that:

1. brings together producers and consumers of FOSS under shared governance to negotiate a fair price for (a) **companies** to pay for FOSS usage, distributed through (b) **ecosystem foundations** to (c) **individual maintainers** on a purpose-built platform we create; and
2. fleshes out and educates the industry on software sharing practices between closed and Open Source.

That's a lot to dump here, so I've written up [a longer case to unpack my thinking](https://openpath.chadwhitacre.com/2024/the-case-for-a-new-institution/).

> The aim is to help educate people on the values of the different models, in a way that has unfortunately become impossible around the topic of general open source.
>
> The overlap with sustainability is primarily from the perspective of how do we keep access to software, to safely sustain its development, rather than resorting to closed source commercial approaches of history.

If _all_ we're doing is [educating](https://github.com/softwarecommons/softwarecommons.com/issues/1) then I guess that's fine, but we should put it up on [howtoshare.software](https://howtoshare.software/) or something like that, we shouldn't call that Software Commons. 2¢

**Comment by mswilson** [comment]
Wow. There's **so much** good stuff to unpack in your blog post. I love how your post gets down to economic first-principles. It recognizes that software is an information good, and that information goods are naturally public goods. There's so much good stuff here. You have a number of great references. I'd suggest also having a look at:

* [Understanding Knowledge as a Commons: From Theory to Practice](https://direct.mit.edu/books/edited-volume/3807/chapter-abstract/124975/Introduction-An-Overview-of-the-Knowledge-Commons?redirectedFrom=fulltext) by  Charlotte Hess and Elinor Ostrom (PDF of introduction chapter is available [here](http://mitp-content-server.mit.edu:18180/books/content/sectbyfn?collid=books_pres_0&fn=9780262083577-Intro.pdf&id=6980))

We can't discuss the nature of Commons without citing Ostrom's research. If you choose to use "Commons" in your branding, the theories and research into Commons will be prominent in the discourse about your ideas and objectives.

There are still-growing social and political movements around advancing the ideas of [Commoning](https://www.onthecommons.org/work/what-commoning-anyway/index.html) that often advocate shifting thinking from "you're on your own" to "we're in this together". I don't think that your goals are aligned with these existing movements.

Movements advancing Commons concepts are concerned about achieving "sustainability" in the system that maintains the resources and goods produced that are central to Commoning, and enables Commoners (those who depend on the Commons) to meet "basic needs for sustenance, economic security, and social connections." You're concerned about "sustainability" as well, but it seems to be focused on "how does a producer of software achieve a _sustainable_ business model."

Bringing producers and consumers of software together to bargain over what a "fair price" is for the information goods that are produced is more the former mindset ("YO-YO") than the latter ("WITT"). That's how the marketplace around information goods works today under established intellectual property rights.

A Commons approach can bring producers and consumers of software together as _peers_. It enables a consumer of software to participate in the production and maintenance of commonly-held information goods, and to share in prosperity when enjoying those goods for their productive uses.

The TL;DR is: I think you absolutely **must** understand your goals and objectives before you settle on branding for an initiative, whether it's an a vendor-oriented website (similar to what "[Fair Source](https://fair.io/)" ended up being?) or establishing a multi-stakeholder initiative. When you hastily adopt terms from existing social and political movements in your branding and marketing, you adopt preexisting values and ideas of those movements. If your values and objectives don't align, you invite criticism from those groups, and confusion from your audience when they a different message from others than what you're trying to deliver.

**Comment by chadwhitacre** [comment]
> [Understanding Knowledge as a Commons: From Theory to Practice](https://direct.mit.edu/books/edited-volume/3807/chapter-abstract/124975/Introduction-An-Overview-of-the-Knowledge-Commons?redirectedFrom=fulltext)

Arriving Tuesday! I ordered it after [this comment](https://github.com/chadwhitacre/openpath/issues/14#issuecomment-1977661604), glad for the +1. 😌

> You're concerned about "sustainability" as well, but it seems to be focused on "how does a producer of software achieve a sustainable business model."

I _have_ been trying to combine these concerns, but I'm coming around to seeing that it's better to keep these two goals separate:

1. Finding language other than Open Source to talk about companies like Sentry who want to share their core products.
1. Solving the [sustainability crisis in Open Source](https://openpath.chadwhitacre.com/2024/the-open-source-sustainability-crisis/).

The former is what this repo and issue originally set out to address, I sort of brought in the other because I care about it so much I guess :), and now I'd like to spin that portion of the conversation off into [a new ticket over in the new `softwarecommons` repo](https://github.com/softwarecommons/softwarecommons.com/issues/4). This current repo has morphed into a focus on FSL, which is fine but is too specific even for that first goal, so I expect to spin up another repo to work on a new educational website (roughly in the vein of choosealicense.com).

I'll likely close this ticket out in the next day or two. 🙏

**Comment by MattiSG** [comment]
> I think you absolutely **must** understand your goals and objectives before you settle on branding for an initiative

Strong +1.

That new outlook at “Software Commons” as a new institution to be created on top directly conflicts with the recurring calls in the last 1.5 years to establish a Foundation for Commons that would be focused on funding and supporting open-source software that has democratic governance and societal impact (so, quite the opposite of “finding sustainable business models in open-source for initial vendors through increased friction to reuse”).

References:

• OpenFuture's [Paradox of Open](https://paradox.openfuture.eu/policies/) ([summary](https://maly.io/@MattiSG/111497868924056051))
• Joint Statement on the [Creation of a European Initiative for Digital Commons](https://www.diplomatie.gouv.fr/en/french-foreign-policy/digital-diplomacy/news/article/joint-statement-by-the-ministry-for-europe-and-foreign-affairs-and-the-state)
• Next Generation Internet: [Forum 2023](https://ngiforum2023.eu) and [Commons call](https://commons.ngi.eu)
• Open Knowledge Foundation’s [Tensions in Commons](https://tensionsincommons.net)
• PublicSpaces [statement](https://publicspaces.net/2023/10/13/public-digital-infrastructure-fund-op-weg-naar-europese-steun-voor-publieke-digitale-infrastructuren/) and [Building Digital Commons](https://video.publicspaces.net/w/gwLJu27zb8a4r1dRi7wdU1?start=13m10s) conference

**Comment by chadwhitacre** [comment]
> the production and maintenance of commonly-held information goods

@mswilson Let's pick this up at https://github.com/softwarecommons/softwarecommons.com/issues/4#issuecomment-1984184686.

> finding sustainable business models in open-source for initial vendors through increased friction to reuse
> References

@MattiSG Let's pick these up at https://github.com/softwarecommons/howtoshare.software/issues/1#issuecomment-1984148621 and https://github.com/softwarecommons/softwarecommons.com/issues/4#issuecomment-1984150708.

**Comment by chadwhitacre** [comment]
I think that's it? Closing again. 🙈

Thank you all for your participation! Epic thread! 😅 🙏 💃

**Comment by chadwhitacre** [comment]
Hey all, circling back to this old thread to share that we launched [Fair Source](https://fair.io/) today.

Revisiting @dcramer's original statement:

> The intent here is to normalize a category of software which provides more freedom than closed source, and more freedom than typical source-available software. It should not diminish the value of FOSS (or OSI's definition of "Open Source"), but should also not be overly confusing to end-users, and not include prejudice.

I'd say we've done a pretty good job of meeting that. Judge for yourself on [fair.io](https://fair.io/).

Cheers! 🍻

**Comment by ssddanbrown** [comment]
@chadwhitacre Congrats, The site looks great! Beautiful, well-organised and easy to read. I like that you defined some level of definition to back the term.

The only part that raised en eyebrow when reading was the text under the "How does Fair Source relate to Open Source?" FAQ item. The text here feels somewhat opinion based from a certain viewpoint than keeping to the functionality of fair source/open source.

> Fair Source provides as many of the benefits of [Open Source](https://opensource.org/) **to developers as companies are comfortable giving**, while reserving and asserting a minimal set of exclusive rights to help achieve a sustainable business model.

**Emphasis mine**. Not sure why this is specific "to developers" since everyone (including end-users, companies) would gain from those shared benefits. The "companies are comfortable giving" really depends on the company and their viewpoint/goals. I feel the line would work better without this. Also, linking "Open Source" in that to https://opensource.org/osd may be better as to provide the context of what open source is, rather than what the OSI is.

Then in regard to the two boxes beneath this:

> Fair Source works best for a company’s core software products. 
> Open Source is best for infrastructure shared across companies. 

That's quite an opinionated take which again depends on perspective, goals, scenario & ideals.

As an extra note, the last FAQ item makes it sound like you're assuring that all fair source code will be safe (in terms of security) to use. I get what you're going for though but haven't been able to think of a better succinct title myself.

**Comment by chadwhitacre** [comment]
Thanks for the feedback and kind words, @ssddanbrown! Sorry it's taken a while to circle back. I've addressed your points in https://github.com/fairsource/fair.io/pull/54. 👍
